src/main/java/CommandParser.java

```diff
@@ -0,0 +1,80 @@
+public class CommandParser {
+    private Commands flag;
+    private String parameter;
+
+    public CommandParser(){
+        flag = Commands.UNKNOWN;
+        parameter = Constants.ZERO_LENGTH_STRING;
+    }
+
+    public void parse(String rawInput){
+        if (rawInput == null) return;
+        String[] input = rawInput.split(Constants.SPACE);
+        updateFlag(input[0]);
+        if (flag == Commands.DEADLINE || flag == Commands.EVENT ||
+                flag == Commands.DONE || flag == Commands.UNDONE){
```

Comment:
This section might be a bit too long for an argument in if() statement and the flag might not be indicative of what it means.

Perhaps you can create a new boolean flag to replace the arguments in if() function and have it named differently to indicate what the flag indicate?